pkg/alertmanager/alertmanager_ring.go

```diff
@@ -79,6 +81,7 @@ func (cfg *RingConfig) RegisterFlags(f *flag.FlagSet) {
 	cfg.KVStore.RegisterFlagsWithPrefix(rfprefix, "alertmanagers/", f)
 	f.DurationVar(&cfg.HeartbeatPeriod, rfprefix+"heartbeat-period", 15*time.Second, "Period at which to heartbeat to the ring. 0 = disabled.")
 	f.DurationVar(&cfg.HeartbeatTimeout, rfprefix+"heartbeat-timeout", time.Minute, "The heartbeat timeout after which alertmanagers are considered unhealthy within the ring. 0 = never (timeout disabled).")
+	f.DurationVar(&cfg.FinalSleep, rfprefix+"final-sleep", 30*time.Second, "The sleep seconds when alertmanager is shutting down.")
```

Comment:
Is this flag useful mainly for migration between Memberlist to DDB? And not used in other cases?
If it is only useful for a specific scenario, I would prefer to set the default value to 0s, which doesn't change existing behavior

Reply:
for memberlist this effect is small, because the information can reach other nodes within 2 secs. But for DDB, it takes 30s, we need to sleep to wait other pods know this pod is bad.
@danielblando what do you think? A default zero?

Reply:
We do have other ring implementations, too. Not only DDB and memberlist.
The value should take all as considerations.
To not break existing cases, we should do 0s

Reply:
Same as before. Just set default to 0. So default behavior is same as before.